### ISSUE TITLE
"@..." mentions sent from the CLI are not clickable in the Slack app

### ISSUE THREAD

**Issue by BoD** [regisb/slack-cli#23]
Hi!  Sorry, if this is not really a bug but simply something I don't know how to do.

When I'm mentionning somebody in the official Slack client (for instance @username) this becomes clickable in its UI.

But when using `slack-cli` (for instance `slack-cli -t xxx -d mychannel "Hello, @bod"`) they are not.  Is it a known issue / something that could be fixed?

Thanks a lot!

**Comment by regisb** [comment]
Hi @bod! Unfortunately, in a shell only urls are clickable (at best). I'm afraid we are limited by the features of shells here...
I'm going to close this issue, but feel free to reopen if you have an idea on how to implement this.

**Comment by BoD** [comment]
Thanks for replying. I think maybe I wasn't clear enough. I wasn't talking about names being clickable in the shell, but in the official Slack app.

Basically, if I send a message through the official app, names are clickable (in the official app), but if I send the same message through the cli, they are not (also in the official app). 

So it must be a difference in the way messages are sent?

**Comment by regisb** [comment]
Ho, right, I understand. This is a legitimate issue. I'll re-open and look into it.

**Comment by nehemiahjacob** [comment]
So when a @user is mentioned and selection in the official app, the text message goes line this
```
{
  "ok": true,
  "message": {
    "client_msg_id": "c7f1f455-7843-4e1d-9f89-281b40874b1a",
    "type": "message",
    "text": "Hello <@U7ECCNDN1>"
  }
}
```

So right before POSTing the message to the server, we have to replace @user into <@U&..> I would like to work on this item if you give me some pointers.

**Comment by regisb** [comment]
Hey @nehemiahjacob, thanks for investigating this issue and your offer to contribute! You will need to modify the `slackcli.slack.post_message` method. You will want to search and replace "@" mentions by looking for the proper user id. Currently, the `slackcli.names.username` method performs the username -> user ID association, but there is no method to do the association in the opposite direction.

I don't know how familiar you are with Python? Please don't be afraid to draft a quick'n dirty pull request, I'll comment on it ;)

**Comment by regisb** [comment]
This is now included in v2.2.8.
